tests/Http/CheckTypeOfOfferMiddlewareTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http;
+
+use CultuurNet\UDB3\Http\Request\Psr7RequestBuilder;
+use CultuurNet\UDB3\ReadModel\DocumentDoesNotExist;
+use CultuurNet\UDB3\ReadModel\InMemoryDocumentRepository;
+use CultuurNet\UDB3\ReadModel\JsonDocument;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class CheckTypeOfOfferMiddlewareTest extends TestCase
+{
+    private const EVENT_ID = 'event-f2ab-4317-a3bd-6f8bd656a9d1';
+    private const PLACE_ID = 'place-f2ab-4317-a3bd-6f8bd656a9d1';
+
+    /**
+     * @dataProvider offerTypeProvider
+     */
+    public function testProcessMiddleware(string $offerType, string $offerId, bool $expectException): void
+    {
+        $request = (new Psr7RequestBuilder())
+            ->withUriFromString(sprintf('/%s/%s', $offerType, $offerId))
+            ->withRouteParameter('offerType', $offerType)
+            ->withRouteParameter('offerId', $offerId)
+            ->build('GET');
+
+        $handler = $this->createMock(RequestHandlerInterface::class);
+
+        $placeRepository = new InMemoryDocumentRepository();
+        $placeRepository->save(new JsonDocument(self::PLACE_ID, '{}'));
+
+        $eventRepository = new InMemoryDocumentRepository();
+        $eventRepository->save(new JsonDocument(self::EVENT_ID, '{}'));
+
+        $middleware = new CheckTypeOfOfferMiddleware($placeRepository, $eventRepository);
+
+        if ($expectException) {
```

Comment:
The happy paths don't have an assert or expect so you introduced risky tests, see https://github.com/cultuurnet/udb3-backend/actions/runs/7627184604/job/20775262612?pr=1584#step:8:91

Reply:
Problem is I don't really know what to test here. Before I mocked the EventRepo and placeRepo, and then I had ->expects($this->once()) and the correct repo, so at least I tested that.
I lost that because I now removed the mock.

I can reintroduce the mock if you want?

Reply:
I think there is no need to test the happy paths, those are handled in various other unit tests

Reply:
Removed the happy paths